src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -116,8 +119,21 @@ impl<T: 'static> EventLoop<T> {
         let output_manager = OutputManager::new(&env);
 
         // A source of events that we plug into our event loop.
-        let wayland_source = WaylandSource::new(event_queue).quick_insert(event_loop.handle())?;
-        let wayland_source = Rc::new(wayland_source);
+        let wayland_source = WaylandSource::new(event_queue);
+        let wayland_dispatcher = calloop::Dispatcher::new(wayland_source, |_, queue, ddata| {
+            queue.dispatch_pending(ddata, |event, object, _| {
+                panic!(
+                    "[calloop] Encountered an orphan event: {}@{} : {}",
+                    event.interface,
+                    object.as_ref().id(),
+                    event.name
+                );
```

Comment:
I'd be curious what this means exactly, I'd assume this happens when a callback isn't defined for an event? Is there a need to panic here?

Reply:
It's for the case when wayland object isn't attached to any queue but you get a callback for it, with basically shouldn't happen, and when it's happening it's a bug in winit logic.

Reply:
Does that bug in winit logic indicate that things will never be working again properly in the future anyway? Otherwise I think a panic isn't ideal.